### PR TITLE
REP-2162 Response Status Code BUG

### PR DESCRIPTION
**Pull request by wdschei**
Added tests for several filters to ensure the Response Status Code is being set appropriately and fixed those were it wasn't.

Squashed the OpenStackResponseHandler into the only use of the class to alleviate unnecessary abstractions.

Updated the TranslationHandler to not have duplicated logic with an embedded return.

### CODE REVIEW

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHandler.java [hunk 1]
```diff
@@ -26,15 +26,20 @@
 import org.openrepose.common.auth.openstack.AuthenticationService;
 import org.openrepose.common.auth.openstack.AuthenticationServiceClient;
 import org.openrepose.common.auth.openstack.OpenStackToken;
+import org.openrepose.commons.utils.StringUtilities;
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHandler.java]: Don't use this string utilities use the one from commons lang

Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHandler.java]: Oh, I see you just collapsed something in, eh, fix it anyway.

File: repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java [hunk 2]
```diff
@@ -81,7 +82,7 @@ public void shouldReturnUnauthorizedWhenUserInformationIsMissing() {
 
         @Before
         public void setup() {
-            List<String> headerNames = new ArrayList<String>();
+            final List<String> headerNames = new ArrayList<>();
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: why the addition of final?

File: repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java [hunk 3]
```diff
@@ -93,7 +94,12 @@ public void setup() {
             defaultConfig.setValue(10);
             defaultConfig.setUnit(org.openrepose.core.services.ratelimit.config.TimeUnit.MINUTE);
 
-            when(mockedRequest.getHeaderNames()).thenReturn(Collections.enumeration(headerNames));
+            when(mockedRequest.getMethod()).thenReturn("GET");
+            when(mockedRequest.getHeaderNames()).thenAnswer(new Answer<Object>() {
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: Why this change?

File: repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java (hunk 2 above)
Comment (wdschei) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: Since the `headerNames` variable is now accessed directly in the anonymous `Answer` class it has to be declared `final`.

File: repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java (hunk 3 above)
Comment (wdschei) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: The original declaration of the mock'd when/thenReturn of the `headerNames` in the `setup()` was actually non-functional. This is why at the top of each method there was another re-declaration of this when/thenReturn clause. I just pulled working when/thenReturn from the top of each method and placed it in the `setup()` in turn forcing the `final` declaration of the `headerNames`.

Comment (adrianjgeorge) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: Why is answer necessary at all? you should be able to get then return to work. what error is it giving?

Comment (wdschei) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: It is not giving an error, the returned enumeration is just empty.

Comment (wdschei) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: This seems to be by it was copy/pasted into each of the tests.

Comment (adrianjgeorge) [inline repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java]: that's super weird

**Comment by wdschei** [comment]
Retest this please.

**Comment by adrianjgeorge** [comment]
seems to be legit
